solr/core/src/java/org/apache/solr/response/transform/ChildDocTransformerFactory.java

```diff
@@ -160,8 +161,18 @@ protected static String processPathHierarchyQueryString(String queryString) {
     int indexOfLastPathSepChar = queryString.lastIndexOf(PATH_SEP_CHAR, indexOfFirstColon);
     if (indexOfLastPathSepChar < 0) {
       // regular filter, not hierarchy based.
-      return ClientUtils.escapeQueryChars(queryString.substring(0, indexOfFirstColon))
```

Comment:
@dsmiley I need your help on this!  I am struggling on childFilters...   `level_i:1` works great with the escaping, but queries like `level_i:[1 TO 3]` fail due to the escaping, I get werid "can parse number" errors.   Same if I have `type_s:Chocolate OR type_s:Regular`, but if I skip that escaping then things work.   

This is definitly at the very outer edges of my knowledge. Would love to chat about how to fix this.

Reply:
I think there should **not** have been escaping here in the first place.  Unless the user is using the special syntax, we should handle it plainly like we do everywhere else -- simple.  It may mean the user has to escape characters like '/' but that's normal.  In a Java String that has an embedded local-params for the child doc transformer, that winds up being four back-slashes.  So be it.  It may seem like we're trying to do the user a favor by escaping what we see but I think it's actually more confusing to reason about because it's inconsistent.  Besides, there are other ways of constructing the query syntax that results in zero escaping.  like `childFilter='{!field f=_nest_path_}/toppings'`  So that's longer but no escaping at least.  Up to the user's prerogative.
CC @moshebla

I'll fix this in your PR momentarily.